src/index.ts

```diff
@@ -10,10 +10,10 @@ type State = {
   web3Modal?: Web3Modal;
 };
 
-const useStore = create<State>(_set => ({}));
+const useStore = create<State>(_set => ({ web3Modal: new Web3Modal() }));
```

Comment:
It wasn't clear to me why you were doing this inside a `useEffect` (which would trigger an immediate re-render) if we can just set this up here.